starknet_devnet/transactions.py

```diff
@@ -52,8 +52,10 @@ def __get_actual_fee(self) -> int:
 
     def __get_events(self) -> List[Event]:
         """Returns the events"""
-        contract_address = self.execution_info.call_info.contract_address
-        return [Event.create(event_content=e, emitting_contract_address=contract_address) for e in self.execution_info.call_info.events]
+        if isinstance(self.execution_info, StarknetTransactionExecutionInfo):
```

Comment:
Which other class can it be other than `StarknetTransactionExecutionInfo`? And when is it so? Why doesn't that other class support using `raw_events`?

Reply:
What other type it can be is defined by in the `execution_info` type. It can be either `TransactionExecutionInfo` for invokes or `StarknetTransactionExecutionInfo` for deploys

Reply:
That's the only `if`. We could transform `TransactionExecutionInfo` to `StarknetTransactionExecution` but I made a priority of fixing this instead

Reply:
[16d3eaf](https://github.com/Shard-Labs/starknet-devnet/pull/118/commits/16d3eaf20b0ef17d14e901f305b906d861ea03d0)